src/Sarif/ExtensionMethods.cs

```diff
@@ -75,7 +75,7 @@ public static string InstanceIdLogicalComponent(this RunAutomationDetails runAut
 
             if (!instanceId.Contains("/"))
             {
-                return String.Empty;
+                return null;
```

Comment:
>null [](start = 23, length = 4)

This is what cause the empty `stableId` that our customer saw. Just omit it.

Reply:
When are we going to get started on unit testing? This is a spot fix, I'd expect a spot unit test. Using the functional test harness is a getting a bit heavyweight for this sort of detail bug fix. For this case, for example, we could have simply tested/documented the purpose and behavior of this method in a dedicated unit test. Something to keep in mind moving forward. I don't want to see us add multiple files of tests for spot fixes, this gets very burdensome over time.

---
In reply to: [299228991](https://github.com/microsoft/sarif-sdk/pull/1558#discussion_r299228991) [](ancestors = 299228991)